docs/libvirt-node-networking.md

```diff
@@ -0,0 +1,298 @@
+# Layer Two Connectivity for nested k8s Nodes
+
+The purpose of this proposal, is to allow connecting VMs to the node network,
+in contrast to
+[docs/libvirt-pod-networking.md](docs/libvirt-pod-networking.md), which allows
+VMs to be connected to the Kubernetes Overlay network. This allows VMs started
+by KubeVirt, to act as full k8s nodes themselves. Inside the VMs, the overlay
+network provider used in the k8s installation will work as expected and the
+nested node looks like an ordinary node from networking perspective.
+
+There are three main points to consider, to achive this:
+
+ 1) How and where to specify what the master host network device to connect to
+    is.
+ 2) How can the nested nodes and the real node communicate with each other.
+ 3) How to bring a device in position, so that libvirt can use it.
+ 
+ Further the proposal focuses on two network topologies:
+ 
+ 1) The master host network device is **not** a **bridge**
```

Comment:
I found the distinction here is misleading. We have two options to connect the host device. One is to add a bridge (during deployment, or with a special "day 2" tool). The other is to add a macvlan/macvtap pair.

We do not discuss passthrough of host devices (such as SR-IOV virtual functions).